pkg/pma/pma.h

```diff
@@ -0,0 +1,150 @@
+#ifndef PMA_PMA_H
+#define PMA_PMA_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+//==============================================================================
+// TYPES
+
+/// Status of a page in a PMA.
+enum page_status {
+    /// Page hasn't been read or written.
+    PS_UNMAPPED = 0x0,
+    /// Page has been read but not written.
+    PS_MAPPED_CLEAN = 0x1,
+    /// Page has been written.
+    PS_MAPPED_DIRTY = 0x2,
+    /// Page is mapped but can't be read or written.
+    PS_MAPPED_INACCESSIBLE = 0x3,
+    PS_MASK                = 0x3,
+};
+typedef enum page_status page_status_t;
+
+/// Get the active length of the heap and stack.
```

Comment:
```suggestion
/// Get the in-memory length of the heap and stack.
```

I think replacing "active" with "in-memory" is more clear (if it's correct of course).

Reply:
I agree active is a little vague, but in-memory isn't quite right. This callback allows the caller to define the heap and stack length that should be persisted; these lengths are less than the in-memory lengths of the heap and stack, respectively.

Reply:
Ah, I see. Maybe something like "Get the lengths of the heap and stack that should be persisted (to the [file backing | memory])." Does that sound better?